### DISCUSSION TITLE
Carbon type casting in Route parameter returns a fresh instance of Carbon

### DISCUSSION THREAD

**Discussion by robguy21** [laravel/framework#51416] [Ideas]
### Laravel Version

11

### PHP Version

8.4

### Database Driver & Version

-

### Description

With the following Route handler:
```
Route::get('/test/{ymd_date}', function (\Carbon\Carbon $ymd_date) {
    dd($ymd_date);
});
```

The following API requests return the below-
```
GET /test/2024-04-04

Carbon\Carbon ...
  date: 2024-05-14 22:25:14.854722 UTC (+00:00)
```

I would expect the argument $ymd_date to return a carbon instance from '2024-04-04'.

This is a bit confusing as a user may expect this behaviour to work given how Model inject as expected.

### Steps To Reproduce

1) Init a new project

2) Create a new controller with a typed-Carbon argument

3) Make a request to that endpoint

4) Note a fresh instance of carbon is created, it does not use the argument in the url

**Comment by countnoobula** [comment]
I have the exact same issue to this. I have tried to look through the IoC tonight, but failed to make any progress on it. 
I did also try use `Illuminate\Support\Carbon` as Laravel seems to have taken it into the core, but still to no success.

I'll see if I can whip up a PR for this soon when I have some time
